### PR TITLE
Added implementation for SuppressTokensFilter

### PR DESCRIPTION
**Pull request by jkrukowski**
- added implementation for SuppressTokensFilter
- added tests

If this PR looks good, I can try to implement other filters 🫡

### CODE REVIEW

**Comment by ZachNagengast** [comment]
Awesome contribution! Reviewing this shortly

File: Tests/WhisperKitTests/UnitTests.swift [hunk 1]
```diff
@@ -447,10 +447,62 @@ final class UnitTests: XCTestCase {
...
+            var linearOffset = 0
+            for (dimension, stride) in zip(index as [NSNumber], strides) {
+                linearOffset += dimension.intValue * stride.intValue
+            }
+            result.append(ptr[linearOffset])
+        }
+        return result
+    }
+}
+
```
Comment (ZachNagengast) [inline Tests/WhisperKitTests/UnitTests.swift:505]: Nice 👍

**Review by ZachNagengast — CHANGES_REQUESTED** [review]
This looks great @jkrukowski just a couple small notes, will run through the full tests as well.

**Review by ZachNagengast — APPROVED** [review]
Updates look good, everything is passing, well done!

**Comment by ZachNagengast** [comment]
Would love to see your take on the other filters if you're up for it! I'm going to be looking at word-level timestamps next on my side FYI, so we'll definitely need timestamp filtering i.e. [`ApplyTimestampRules`](https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/decoding.py#L441) equivalent for us, but keep in mind some of this logic we pulled into the `SegmentSeeking` protocol.

**Comment by jkrukowski** [comment]
> Would love to see your take on the other filters if you're up for it! I'm going to be looking at word-level timestamps next on my side FYI, so we'll definitely need timestamp filtering i.e. [`ApplyTimestampRules`](https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/decoding.py#L441) equivalent for us, but keep in mind some of this logic we pulled into the `SegmentSeeking` protocol.

sure, happy to do it. Will try to find some time this or next week

**Comment by jkrukowski** [comment]
> Would love to see your take on the other filters if you're up for it! I'm going to be looking at word-level timestamps next on my side FYI, so we'll definitely need timestamp filtering i.e. [`ApplyTimestampRules`](https://github.com/openai/whisper/blob/ba3f3cd54b0e5b8ce1ab3de13e32122d0d5f98ab/whisper/decoding.py#L441) equivalent for us, but keep in mind some of this logic we pulled into the `SegmentSeeking` protocol.

hi @ZachNagengast, not sure if it's a good place to ask but you mentioned that some of this logic was pulled into `SegmentSeeking` protocol. Can you please elaborate a bit more?

**Pull request merged**